### ISSUE TITLE
About `decoder_input_ids` in BART doc

### ISSUE THREAD

**Issue by ShaneTian** [huggingface/transformers#15691]
> For translation and summarization training, decoder_input_ids should be provided. If no decoder_input_ids is provided, the model will create this tensor by shifting the input_ids to the right for denoising pre-training following the paper.
https://huggingface.co/docs/transformers/v4.16.2/en/model_doc/bart#transformers.BartForConditionalGeneration.forward.decoder_input_ids

This is inappropriate. If `labels` is provided, the model will create this tensor by shifting the `labels`:
https://github.com/huggingface/transformers/blob/db7d6a80e82d66127b2a44b6e3382969fdc8b207/src/transformers/models/bart/modeling_bart.py#L1320-L1324

**Comment by NielsRogge** [comment]
Indeed, similar to #11357 and duplicate of #14328.

Can you open a PR to fix this? Otherwise I'll do it.

**Comment by ShaneTian** [comment]
I have submit a PR to fix it. Can you review it?
